### PR TITLE
add openshift usecase

### PR DESCRIPTION
**Pull request by chanwit**
Signed-off-by: Chanwit Kaewkasi <chanwit@gmail.com>

### CODE REVIEW

File: content/en/docs/use-cases/openshift.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
+---
+title: Using Flux on OpenShift
+linkTitle: OpenShift
+---
+
+## OpenShift Setup
+
+Steps described in this document has been tested on OpenShift 4.6 only.
```
Comment (dholbach) [inline content/en/docs/use-cases/openshift.md:8]:
```suggestion
Steps described in this document have been tested on OpenShift 4.6 only. 
```

File: content/en/docs/use-cases/openshift.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
+---
+title: Using Flux on OpenShift
+linkTitle: OpenShift
+---
+
+## OpenShift Setup
+
+Steps described in this document has been tested on OpenShift 4.6 only. 
+You require cluster-admin privileged to install Flux on OpenShift. This means that it currently is not possiple to install Flux on the OpenShift Developer Sandbox.
```
Comment (dholbach) [inline content/en/docs/use-cases/openshift.md:9]:
```suggestion
You require cluster-admin privileges to install Flux on OpenShift. This means that it currently is not possible to install Flux on the OpenShift Developer Sandbox.
```

Comment (dholbach) [reply content/en/docs/use-cases/openshift.md:9]: Does this need to be explained, e.g. what OpenShift Developer Sandbox is or why it's not possible, or is this clear to OS users?

File: content/en/docs/use-cases/openshift.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+### CodeReady Containers
+An easy way to provision OpenShift is to use CodeReady Containers (CRC) for OpenShift, which could be obtained from [here](https://developers.redhat.com/products/codeready-containers/overview). With this setup, you require a physical Linux box. An OpenShift cluster will run inside a VM installed by CRC. 
+
+{{% alert %}}
+Despite being a single node, the cluster will consume a large amount of both CPU and memory resources. Your desktop / laptop must have at least 8 CPU cores with 32 GB of RAM as it is recommened to allocate at least 4 CPU cores and 18 GB of RAM for the cluster to have a good experience.
```
Comment (dholbach) [inline content/en/docs/use-cases/openshift.md:15]:
```suggestion
Despite being a single node, the cluster will consume a large amount of both CPU and memory resources. Your desktop / laptop must have at least 8 CPU cores with 32 GB of RAM as it is recommended to allocate at least 4 CPU cores and 18 GB of RAM for the cluster to have a good experience.
```

File: content/en/docs/use-cases/openshift.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+{{% /alert %}}
+
+```sh
+# Setup the OpenShift configuration
+# You require to paste a pull secret here
```
Comment (dholbach) [inline content/en/docs/use-cases/openshift.md:20]:
```suggestion
# You need to paste a pull secret here
```

File: content/en/docs/use-cases/openshift.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+crc setup
+
+# Start the cluster with 18 GB of RAM
+crc start -c 4 -m 18432
+```
+
+After the cluster up and running, there will be a message tell us how to login.
```
Comment (dholbach) [inline content/en/docs/use-cases/openshift.md:27]:
```suggestion
After the cluster is up and running, there will be a message tell us how to login.
```

File: content/en/docs/use-cases/openshift.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+# Prepare environment setup for the OC command
+eval $(crc oc-env)
+
+# Login 
+oc login -u kubeadmin -p <your password> https://api.crc.testing:6443
+```
+
+### Security Context Constraints
+
+Before installing Flux, you require to set the **privileged** security context constraint for the following controller in the `flux-system` namespace.
```
Comment (dholbach) [inline content/en/docs/use-cases/openshift.md:40]:
```suggestion
Before installing Flux, you need to set the **privileged** security context constraint for the following controller in the `flux-system` namespace.
```

File: content/en/docs/use-cases/openshift.md [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+NS="flux-system"
+oc adm policy add-scc-to-user privileged system:serviceaccount:$NS:source-controller
+oc adm policy add-scc-to-user privileged system:serviceaccount:$NS:kustomize-controller
+oc adm policy add-scc-to-user privileged system:serviceaccount:$NS:image-automation-controller
+oc adm policy add-scc-to-user privileged system:serviceaccount:$NS:image-reflector-controller
+```
+
+## Flux Installation with CLI
+
+The best way to install Flux on OpenShift currently is to use `flux bootstrap` command. Assuming that you are a GitHub user, you could start by preparing your GitHub credentials.
```
Comment (dholbach) [inline content/en/docs/use-cases/openshift.md:52]:
```suggestion
The best way to install Flux on OpenShift currently is to use the `flux bootstrap` command. Assuming that you are a GitHub user, you could start by preparing your GitHub credentials.
```

File: content/en/docs/use-cases/openshift.md [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+  --branch=main \
+  --path=./clusters/my-cluster \
+  --personal
+```
+
+and enjoy your GitOps on OpenShift.
+
+## Flux Upgrade
+
+Upgrading Flux on OpenShift is very simple and straightforward. Please just make sure that you already logged in as `kubeadmin` user. Assuming you are a Homebrew user, you could upgrade Flux CLI using the following command.
```
Comment (dholbach) [inline content/en/docs/use-cases/openshift.md:74]:
```suggestion
Upgrading Flux on OpenShift is very simple and straightforward. Please just make sure that you are already logged in as `kubeadmin` user. Assuming you are a Homebrew user, you could upgrade Flux CLI using the following command.
```

File: content/en/docs/use-cases/openshift.md [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+brew upgrade fluxcd/tap/flux
+
+# Check Flux version
+flux -v
+
+# Login as kubeadmin
+oc login -u kubeadmin -p <your password> https://api.crc.testing:6443
+```
+
+After you obtained the Flux version you wanted, simply re-run the above `flux bootstrap` command from the prevous section, and all of your Flux component will be upgraded.
```
Comment (dholbach) [inline content/en/docs/use-cases/openshift.md:87]:
```suggestion
After you obtained the Flux version you wanted, simply re-run the above `flux bootstrap` command from the previous section, and all of your Flux component will be upgraded.
```

**Review by dholbach — COMMENTED** [review]
Generally :+1: - just a few small typos.

**Review by stefanprodan — COMMENTED** [review]
@chanwit please add links to the install doc for bootstrap and upgrade. Also mention gitlab and the generic git option so people ending up on this page don’t assume flux only works with github.

**Comment by chanwit** [comment]
> @chanwit please add links to the install doc for bootstrap and upgrade. Also mention gitlab and the generic git option so people ending up on this page don’t assume flux only works with github.

fixed.
could you PTAL?

**Review by dholbach — APPROVED** [review]

**Pull request merged**